addons/stock_delivery/models/delivery_carrier.py

```diff
@@ -199,18 +200,19 @@ def _get_commodities_from_order(self, order):
 
     def _get_commodities_from_stock_move_lines(self, move_lines):
         commodities = []
-        for line in move_lines.filtered(lambda line: line.product_id.type in ['product', 'consu']):
-            qty = line.qty_done if line.state == 'done' else line.reserved_uom_qty
-            unit_quantity = line.product_uom_id._compute_quantity(qty, line.product_id.uom_id)
+
+        product_lines = move_lines.filtered(lambda line: line.product_id.type in ['product', 'consu'])
+        for product, lines in groupby(product_lines, lambda x: x.product_id):
```

Comment:
try to not use x as lambda variable. Try something more representative. E.g ml for move_lines here